amp-camp/src/server/server.js

```diff
@@ -312,4 +331,22 @@ function enableCors(req, res) {
     res.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
     res.header("Access-Control-Expose-Headers", "AMP-Access-Control-Allow-Source-Origin,AMP-Redirect-To");
     res.header("AMP-Access-Control-Allow-Source-Origin", sourceOrigin);
+}
+
+// Uses mustache to render the dynamic page. Injects canonical tag, based on the requested URL.
+function renderPage(req, res, template, responseJsonObj) {
+    if(!responseJsonObj) {
+        responseJsonObj = {}
+    }
+    
+    let fullUrl = req.protocol + '://' + req.get('host') + req.originalUrl;
+    responseJsonObj.CanonicalLink = fullUrl;
+
+    mustache.tags = ['<%', '%>'];
+    res.render(template, responseJsonObj);
+}
+
+//The root (home) uses index.html as template. In any other case, the url contains the name of the template.
```

Comment:
That's helpful! Can you continue this comment to then describe what logic is being done below? I assume we're building the name of the template file from the URL...